tests/RobotFramework/libraries/ThinEdgeIO/ThinEdgeIO.py

```diff
@@ -962,6 +962,67 @@ def delete_smartrest_one_template(self, template_id: str):
                 ex,
             )
 
+    @keyword("List Entities")
+    def list_entities(
+            self, name: str = None 
+    ) -> List[Dict[str, Any]]:
+        """
+        Get entity list from the device using the entity store query REST API
+
+        Args:
+            name (str, optional): Device name to fetch the entity list from
```

Comment:
I'm having second thoughts about this parameter. Actually, this entity store service or the HTTP server itself is only available on the main device. Even if `tedge-agent` is installed on a child device, these endpoints won't be available. So, accepting this parameter is pointless. In fact, it should be restricted to the main device only. How can I do that?

Reply:
Though it is needed as the api can be reachable from other devices, so we still need it.

It would be very easy to detect if a child device is being used and switch to the correct hostname/ip address instead of using localhost. When the tedge-agent is running on a child device it needs to know the main device hostname/ip address, so it should already have the info needed.

Reply:
But we can live it simple for now and keep the `name` (device name) parameter. The test will just fail if the user tries to run the keyword on a child device (which would still be technically correct ;))

Reply:
Leaving it as-is.